### PR TITLE
feat: pectra

### PR DESCRIPTION
**Pull request by rkrasiuk**
## 📝 Summary

Support for the upcoming Pectra hardfork. Depends on https://github.com/alloy-rs/alloy/pull/1303

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -74,4 +74,9 @@ alloy-rpc-types-engine = { version = "0.3.0", features = [
...
 ] }
 alloy-rpc-types-eth = { version = "0.3.0" }
-alloy-signer-local = { version = "0.3.0" }
\ No newline at end of file
+alloy-signer-local = { version = "0.3.0" }
+
+[patch.crates-io]
+alloy-eips = { git = "https://github.com/alloy-rs/alloy", branch = "rkrasiuk/bid-submission-v4" }
+alloy-rpc-types-beacon = { git = "https://github.com/alloy-rs/alloy", branch = "rkrasiuk/bid-submission-v4" }
+alloy-rpc-types-engine = { git = "https://github.com/alloy-rs/alloy", branch = "rkrasiuk/bid-submission-v4" }
```
Comment (ryanschneider) [inline Cargo.toml]: These can all be `version = "0.3.6"` now?

https://github.com/alloy-rs/alloy/releases/tag/v0.3.6

```suggestion
alloy-eips = { version = "0.3.6" }
alloy-rpc-types-beacon = { version = "0.3.6" }
alloy-rpc-types-engine = { version = "0.3.6" }
```

edit: but shouldn't be a patch anymore, just updated versions above so don't apply the suggestion as-is.

Comment (rkrasiuk) [reply Cargo.toml]: @ryanschneider the spec is being changed, so i'm holding off moving it from draft https://github.com/ethereum/execution-apis/pull/587

**Comment by rkrasiuk** [comment]
@dvush @ferranbt @ryanschneider let's merge this in, i'll come back w/ any updates in case of spec changes

**Review by ryanschneider — APPROVED** [review]
lgtm!

**Review by dvush — APPROVED** [review]

**Pull request merged**